src/main/java/org/broadinstitute/hellbender/utils/downsampling/PositionalDownsampler.java

```diff
@@ -0,0 +1,146 @@
+     * @param header SAMFileHeader to use to determine contig ordering. Non-null.
+     */
+    public PositionalDownsampler( final int targetCoverage, final SAMFileHeader header ) {
+        Utils.validateArg(targetCoverage > 0, "targetCoverage must be > 0");
+        Utils.nonNull(header);
+
+        this.reservoir = new ReservoirDownsampler(targetCoverage);
+        this.finalizedReads = new ArrayList<>();
+        this.header = header;
+        clearItems();
+        resetStats();
+    }
+
+    @Override
+    public void submit( final GATKRead newRead ) {
+        // If we've moved to a new position, finalize the reads currently in our reservoir.
+        handlePositionalChange(newRead);
+
+        // Pass-through reads that have no assigned position, to avoid downsampling all unmapped reads
+        // to the targetCoverage. Unmapped reads that do have an assigned position *will* be subject to
+        // downsampling, however.
+        if ( readHasNoPosition(newRead) ) {
+            finalizedReads.add(newRead);
+        }
+        else {
+            final int reservoirPreviouslyDiscardedItems = reservoir.getNumberOfDiscardedItems();
+            reservoir.submit(newRead);
+            numDiscardedItems += reservoir.getNumberOfDiscardedItems() - reservoirPreviouslyDiscardedItems;
+        }
+
+        lastRead = newRead;
+    }
+
+    private void handlePositionalChange( final GATKRead newRead ) {
+        // Use ReadCoordinateComparator to determine whether we've moved to a new start position.
+        // ReadCoordinateComparator will correctly distinguish between purely unmapped reads and unmapped reads that
+        // are assigned a nominal position.
+        if ( lastRead != null && ReadCoordinateComparator.compareCoordinates(lastRead, newRead, header) != 0 ) {
+            if ( reservoir.hasFinalizedItems() ) {
+                finalizeReservoir();
+            }
+        }
+    }
+
+    private void finalizeReservoir() {
+        finalizedReads.addAll(reservoir.consumeFinalizedItems());
+        reservoir.resetStats();
+    }
+
+    private boolean readHasNoPosition( final GATKRead read ) {
+        // Check actual assigned positions rather than unmapped status, so that unmapped reads with
+        // assigned positions will be processed correctly.
+        return read.getAssignedContig() == null ||
+               read.getAssignedContig().equals(ReadConstants.UNSET_CONTIG) ||
+               read.getAssignedStart() == ReadConstants.UNSET_POSITION;
+    }
+
+    @Override
+    public boolean hasFinalizedItems() {
+        return finalizedReads.size() > 0;
+    }
+
+    @Override
+    public List<GATKRead> consumeFinalizedItems() {
+        final List<GATKRead> toReturn = finalizedReads;
+        finalizedReads = new ArrayList<>();
+        return toReturn;
+    }
+
+    @Override
+    public boolean hasPendingItems() {
+        return reservoir.hasFinalizedItems();
```

Comment:
The finalized items in the `ReservoirDownsampler` are pending items from the perspective of the enclosing `PositionalDownsampler`, since the `PositionalDownsampler` doesn't know how many more reads at the current start position will be submitted to reservoir before we consume its finalized items.